sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -139,28 +132,78 @@ namespace Azure { namespace Security { namespace Attestation {
      *
      * @note TPM attestation requires an authenticated attestation client.
      */
-    explicit AttestationClient(
+    AttestationClient(
         std::string const& endpoint,
         AttestationClientOptions options = AttestationClientOptions())
         : AttestationClient(endpoint, nullptr, options)
     {
     }
 
+  public:
+    /**
+     * @brief Destructor.
+     *
+     */
+    virtual ~AttestationClient() = default;
+
     /** @brief Construct a new Attestation Client object from an existing attestation client.
      *
      * @param attestationClient An existing attestation client.
      */
-    explicit AttestationClient(AttestationClient const& attestationClient)
+    AttestationClient(AttestationClient const& attestationClient)
         : m_endpoint(attestationClient.m_endpoint), m_apiVersion(attestationClient.m_apiVersion),
           m_pipeline(attestationClient.m_pipeline),
           m_tokenValidationOptions(attestationClient.m_tokenValidationOptions){};
 
-    /**
-     * @brief Returns the API version the client was configured with.
+    /** @brief Construct a new Attestation Client object
+     *
+     * @param endpoint The URL address where the client will send the requests to.
+     * @param credential The authentication method to use (required for TPM attestation).
+     * @param options The options to customize the client behavior.
+     */
+    static AttestationClient const Create(
```

Comment:
Is this part of the API review?   Why do we want a factory here?  I haven't seen that elsewhere (on C++)

Reply:
This was a compromise worked out between @tg-msft, @JeffreyRichter  and I - instead of allowing the clients to be explicitly constructed, we use a factory method and issue the magic I/O in the factory method. That was seen as a significant usability barrier during API review (especially since we know exactly what the developer mistake was).

Moving to a factory pattern removes the super-undiscoverable API that you have to call before the service will work. And in addition, since the factory takes a Context, it makes it clear that the factory does I/O.

While the factory pattern is novel to the Attestation service, it's not an unreasonable pattern for C++ classes (for instance, the [C++ core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c50-use-a-factory-function-if-you-need-virtual-behavior-during-initialization) recommend it for certain constructs.